db/dcrpg/internal/vinoutstmts.go

```diff
@@ -58,7 +61,16 @@ const (
 	SelectFundingOutpointByVinID = `SELECT prev_tx_hash, prev_tx_index, prev_tx_tree FROM vins WHERE id=$1;`
 	SelectFundingTxByVinID       = `SELECT prev_tx_hash FROM vins WHERE id=$1;`
 	SelectSpendingTxByVinID      = `SELECT tx_hash, tx_index, tx_tree FROM vins WHERE id=$1;`
-	SelectAllVinInfoByID         = `SELECT * FROM vins WHERE id=$1;`
+	SelectAllVinInfoByID         = `SELECT id, tx_hash, tx_index, tx_tree, is_valid, block_time,
+		prev_tx_hash, prev_tx_index, prev_tx_tree, value_in FROM vins WHERE id = $1;`
+	SetIsValidByTxHash = `UPDATE vins SET is_valid = $1 WHERE tx_hash = $2;`
```

Comment:
This will set for transactions with the same `tx_hash` in orphaned side chains.  The block hash and tx_tree need to be included for this to be operating on only the transaction intended by the caller.

Reply:
ok. Noted

Reply:
Currently vins table doesn't have a block_hash, should I add a block_hash just for this or there is way around it? I think this unique index can work in place of the block_hash..
```go
IndexVinTableOnVins = "CREATE UNIQUE INDEX uix_vin ON vins(tx_hash, tx_index, tx_tree);"
```

Reply:
Ah, yes for the vins table it's ok as is